include/nbl/builtin/hlsl/math/morton.hlsl

```diff
@@ -0,0 +1,423 @@
+    {
+        return !allEqual(rhs);
+    }
+
+    template<class Comparison, class OnSignMismatch>
+    NBL_CONSTEXPR_INLINE_FUNC bool coordOrderCompare(NBL_CONST_REF_ARG(this_t) rhs, uint16_t coord) NBL_CONST_MEMBER_FUNC
+    {
+        NBL_CONSTEXPR_STATIC vector<U, D> Masks = NBL_HLSL_MORTON_MASKS(U, D);
+        Comparison comparison;
+        OnSignMismatch onSignMismatch;
+
+        // When unsigned, bit representation is the same but with 0s inbetween bits. In particular, we can still use unsigned comparison
+        #ifndef __HLSL_VERSION
+        if constexpr (is_unsigned_v<I>)
+        #else
+        if (is_unsigned_v<I>)
+        #endif
+        {
+            return comparison(value & Masks[coord], rhs.value & Masks[coord]);
+        }
+        // When signed, since the representation is unsigned, we need to divide behaviour based on highest bit
+        else
+        {
+            // I will give an example for `operator<` but the same reasoning holds for all others. Some abuse of notation but hopefully it's clear.
+            
+            // If `this[coord] >= 0` and `rhs[coord] < 0` then `this[coord] < rhs[coord]` returns false. Notice that in this case, the highest bit of 
+            // `value` (of the bits representing the number encoded in `coord`) is `0`, while the highest bit for rhs is `1`. 
+            // Similarly, if `this[coord] < 0` and `rhs[coord] >= 0` then `this[coord] < rhs[coord]` returns true, and the highest bit situation is inverted.
+            // This means that if the signs of `this[coord]` and `rhs[coord]` are not equal, the result depends on the sign of `this[coord]`.
+            // What that result should be is controlled by `OnSignMismatch`.
+            // Finally, notice that if only one of those bits is set to 1, then the `xor` of that highest bit yields 1 as well
+            const U highestBit = extractHighestBit(coord);
+            const U rhsHighestBit = rhs.extractHighestBit(coord);
+            if (highestBit ^ rhsHighestBit)
+                return onSignMismatch(highestBit);
+            // If both are nonnegative, then we can just use the comparison as it comes.
+            // If both are negative, it just so happens that applying the same operator to their unsigned bitcasted representations yields the same result.
+            // For `operator<`, for example, consider two negative numbers. Starting from the MSB (we know it's `1` for both in this case) and moving to the right,
+            // consider what happens when we encounter the first bit where they mismatch: the one with a `0` at position `k` (by position I mean counted from the
+            // left, starting at 0) is adding at most `2^k - 1` in the lowest bits, while the one with a `1` is adding exactly `2^k`. This means that the one
+            // with a 0 is "more negative". 
+            else
+                return comparison(value & Masks[coord], rhs.value & Masks[coord]);
```

Comment:
thats not very efficient, the easiest would be to `|(~Mask)` whenever the "synthetic" sign bit is on `extractHighestBit` and use the regular signed comparison operators for type `make_signed_t<storage_t>`